function asyncCallWithPromise(){
    return new Promise(function(resolve, reject){
        if(error)return reject(error)
        resolve(reuslt)
    })
}

asyncCallWithPromise.then(function(result){

}).catch(function(error){
    
})